### DISCUSSION TITLE
What's the fastlane_temp_keychain-db password?

### DISCUSSION THREAD

**Discussion by lukepighetti** [fastlane/fastlane#17002] [Q&A]
I started using GitHub Actions on my local macOS machine as a build runner, it added a keychain `fastlane_tmp_keychain-db` and I have been unable to figure out what the password is. It’s not my `FASTLANE_PASSWORD` or my `MATCH_PASSWORD`. Here's my entire Fastfile.

Does anyone know if there is a default password being used, or what environment variable it is using for the password?

```
default_platform(:ios)
platform :ios do
  desc "Push a new release build to TestFlight"
  lane :testflight do
    setup_ci
    match(
      type: "appstore",
      git_basic_authorization: Base64.strict_encode64(ENV["GIT_BASIC_AUTH"]),
      readonly: true,
    )
    increment_build_number(
      build_number: ENV["GITHUB_RUN_NUMBER"].to_i + 50,
    )
    build_app(
      workspace: "synRadio.xcworkspace",
      scheme: ENV["BUILD_SCHEME"],
    )
    upload_to_testflight(
      app_identifier: ENV["APP_IDENTIFIER"],
      skip_waiting_for_build_processing: true,
    )
  end
end
```

**Comment by tsznokwong** [comment]
I have just encountered this issues these few days. I figured out this temporary keychain should be removed every time so it's always temporary. Try to use the following snippet before and after your TestFlight lane. Remember to remove the keychain when error occurred as well.

```
TEMP_KEYCHAIN = "fastlane_tmp_keychain"

  private_lane :remove_temp_keychain do |options|
    begin 
      next if !is_ci
      name = options[:name].nil? ? TEMP_KEYCHAIN : options[:name]
      delete_keychain(
        name: name
      ) if File.exist? File.expand_path("~/Library/Keychains/#{name}-db")
    rescue => ex
      puts(ex)
    end
  end
```

**Reply by ohthepain** [reply]
This doesn't work for me. The keychain is still there after running the delete_keychain command. I have tried with and without the `-db` at the end of the filename. 

I have no clue why they would make this so complicated. Why not just let us delete the goddamn keychain?

**Comment by crazymanish** [comment]
Hi @lukepighetti 

Here is the code/password for `setup_ci` temporary keychain
https://github.com/fastlane/fastlane/blob/c0b94f1d5b38700b3b1b57031e9ec2931331ac5e/fastlane/lib/fastlane/actions/setup_ci.rb#L48

I hope that will help!

**Reply by JimmyDeemo** [reply]
I too have this problem. Unlocking the keychain doesn't allow me to remove it as a default and my mac is treating it as the first keychain. I'm not familiar with fastlane at all as our build engineer installed it for a new project. How can I remove this temp keychain?

**Comment by crazymanish** [comment]
Hi, if you want to manually delete the keychain `using` **keychain-access**
see here: https://stackoverflow.com/a/68602818/3778686

or `using` lane
see here: https://stackoverflow.com/a/68708714/3778686

**Reply by justingiffard-impact** [reply]
Thank you! `Keychain Access` > Edit > Delete did the trick for me :)

**Comment by okhanokbay** [comment]
This temp keychain is being created by `setup_ci` action of fastlane. And as mentioned in one of the comments above, it is being treated as the default/first/go-to keychain.

Already answered above but as a wrap-up:

**Defined this method:**

```
def cleanup_keychain name
  delete_keychain(name: name) if File.exist? File.expand_path("~/Library/Keychains/#{name}-db")
end
```

**Called it after I'm done with the temp keychain in the Fastfile:**

```
cleanup_keychain("fastlane_tmp_keychain")
```

Seems like working for now.

**Reply by Mas0nSun** [reply]
Thanks, I finally resolved it.
